src/Stratis.Bitcoin.IntegrationTests/NodeSyncTests.cs

```diff
@@ -189,9 +187,9 @@ public void Given_NodesAreSynced_When_ABigReorgHappens_Then_TheReorgIsIgnored()
                 stratisSyncer.CreateRPCClient().RemoveNode(stratisReorg.Endpoint);
                 TestHelper.WaitLoop(() => !TestHelper.IsNodeConnected(stratisReorg));
 
-                Task<List<uint256>> t1 = Task.Run(() => stratisMiner.GenerateStratisWithMiner(11));
-                Task<List<uint256>> t2 = Task.Delay(1000).ContinueWith(t => stratisReorg.GenerateStratisWithMiner(12));
-                Task.WaitAll(t1, t2);
+                stratisMiner.GenerateStratisWithMiner(11);
```

Comment:
How does this fix the test?
IT was done like this for a reason because we will get collisions (hmm unless we use the overlaods)

Reply:
If I run the tests in a task like that it looks like GenerateCoinsFastProvider gets out of sync. I don't understand why we need to run it like this?

Reply:
It was a hack to avoid block collusion,  (I think...)

Reply:
Actually there will be no collusion because it will be done sequentially so I don't  remember why we did it like this maybe @Aprogiena remembers he helped review this.

Reply:
It was certainly needed before just to generate different timestamp. We wanted to different chains to be created and without this we only created one chain by two different miners

Reply:
If you remove the task, you must assert that the two chains are different exactly at the first block that is created here, otherwise you could have false pass

Reply:
@Aprogiena I believe we do check for this a few lines down:

```
 // make sure the nodes are actually on different chains.
Assert.NotEqual(stratisMiner.FullNode.Chain.GetBlock(2).HashBlock, stratisReorg.FullNode.Chain.GetBlock(2).HashBlock);
```

Reply:
But this check means the test will fail.
But I think there is no problem because we are doing it sequential not parallel, it was done in parallel because at the time we mined many more blocks then it was reduced by overriding the maxreorg property I remember now.

Reply:
The test should pass and it needs to pass with those two chains being different